--- bzrlib/transport/http/_urllib2_wrappers.py	2010-10-18 10:01:47 +0000
+++ bzrlib/transport/http/_urllib2_wrappers.py	2010-12-23 19:36:17 +0000
@@ -110,13 +110,8 @@
         self.report_activity(len(s), 'read')
         return s
 
-    def readline(self):
-        # This should be readline(self, size=-1), but httplib in python 2.4 and
-        #  2.5 defines a SSLFile wrapper whose readline method lacks the size
-        #  parameter.  So until we drop support for 2.4 and 2.5 and since we
-        #  don't *need* the size parameter we'll stay with readline(self)
-        #  --  vila 20090209
-        s = self.filesock.readline()
+    def readline(self, size=-1):
+        s = self.filesock.readline(size)
         self.report_activity(len(s), 'read')
         return s

